src/builder.ts

```diff
@@ -95,56 +106,58 @@ export default class Builder {
 							if (data == null) {
 								return;
 							}
-							if (data.error) {
-								errored = true;
-								dup.destroy(new Error(data.error));
-							} else {
-								// Store image layers, so that they can be deleted by the caller
-								// if necessary
-								const sha = Utils.extractLayer(data.stream);
```

Comment:
I am not sure if `extractLayer` could possibly throw an error, but if it did, it wasn't handled. Now there's a try / catch, and a test case for this condition.